### PR TITLE
[NovaAPI]Fix www_authenticate_uri configuration

### PR DESCRIPTION
**Pull request by gibizer**
According to the keystone middleware doc we have to set www_authenticate_uri to the public keystone endpoint as an unauthenticated user is redirected to this URL.

Our service uses the internal endpoint for auth and so far it used the internal endpoint for www_authenticate_uri as well. But this will not work for external users.

A new field is added to the NovaAPI CR to allow configuring not just the internal endpoint but also the public endpoint.

Closes: #217

### CODE REVIEW

**Comment by gibizer** [comment]
/test nova-operator-build-deploy
```
{  failed to wait for the created cluster claim to become ready: timed out waiting for the condition}
```

File: api/bases/nova.openstack.org_novaapis.yaml [hunk 1]
```diff
@@ -155,6 +155,13 @@ spec:
                   type: object
                 type: array
               keystoneAuthURL:
+                description: KeystoneAuthURL configures the keystone API endpoint
+                  to be used by the service for authentication and authorization
+                type: string
+              keystonePublicAuthURL:
```
Comment (SeanMooney) [inline api/bases/nova.openstack.org_novaapis.yaml:161]: dumb question but can we look this up form the keystone catalog?

I'm not nessisarly against adding this just wondering if it is needed

Comment (gibizer) [reply api/bases/nova.openstack.org_novaapis.yaml:161]: We can. We can look for KeystoneEndpoint CRs (how the Nova controller does it to fill this value), or we can use the golang sdk and call keytone internal api to get the catalog and get the public URL.  I went with the former in the nova controller as we already looking for KeystoneEndpoints there.

**Comment by gibizer** [comment]
We know that tempest will only be green again after https://github.com/openstack-k8s-operators/openstack-operator/pull/457 is landed.

**Review by SeanMooney — APPROVED** [review]

**Pull request merged**